test/media-segment-request.test.js

```diff
@@ -173,11 +173,64 @@ QUnit.test('cancels outstanding key requests on timeout', function(assert) {
   assert.equal(keyReq.uri, '0-key.php', 'the first request is for a key');
   assert.equal(segmentReq.uri, '0-test.ts', 'the second request is for a segment');
 
-  keyReq.timedout = true;
   // Timeout request
   this.clock.tick(2000);
 });
 
+QUnit.test('ensure the aborted property is set on aborted requests', function(assert) {
+  let keyReq;
+  const done = assert.async();
+  let oldXHRPrototype = XMLHttpRequest.prototype;
+
+  assert.expect(7);
+  // Sinon automatically adds the aborted property so let's stop it from
+  // doing that to make sure we add it under the expected condition when
+  // running on a real, native XHR object
+  XMLHttpRequest.prototype = Object.create(XMLHttpRequest.prototype);
```

Comment:
Right, thats the point of this test. I mean that with the combination of `xhrFactory` also setting `aborted` to `true` and the reordering of checking for timeouts and aborts before errors sounds like all the other tests that would break by the removal of the line that sets `aborted` to `true` in sinons FakeXHR should not break, and if they are, I feel those tests should be updated to check for the correct error condition